src/lib.rs

```diff
@@ -7,31 +7,33 @@ extern crate quote;
 decl_derive!([Abomonation] => derive_abomonation);
 
 fn derive_abomonation(mut s: synstructure::Structure) -> quote::Tokens {
+
     let entomb = s.each(|bi| quote! {
-        ::abomonation::Abomonation::entomb(#bi, _writer);
+        ::abomonation::Abomonation::entomb(#bi, _write)?;
     });
 
-    s.bind_with(|_| synstructure::BindStyle::RefMut);
-    let embalm = s.each(|bi| quote! {
-        ::abomonation::Abomonation::embalm(#bi);
+    let extent = s.each(|bi| quote! {
+        sum += ::abomonation::Abomonation::extent(#bi);
```

Comment:
I presume extent is not intended to include `size_of::<Self>()`?

Reply:
That's right. It is intended to be "additional bytes, beyond your own". This seems to be what was needed to make it all compose nicely, and is how other methods (`entomb`, `exhume`) are designed.